test/coverlet.core.tests/Coverage/CoverageTest.Yield.cs

```diff
@@ -100,5 +100,65 @@ public void Yield_SingleWithSwitch()
                 File.Delete(path);
             }
         }
+
+        [Fact]
+        public void Yield_Three()
+        {
+            string path = Path.GetTempFileName();
+            try
+            {
+                FunctionExecutor.Run(async (string[] pathSerialize) =>
+                {
+                    CoveragePrepareResult coveragePrepareResult = await TestInstrumentationHelper.Run<Yield>(instance =>
+                    {
+                        foreach (var _ in instance.Three()) ;
+
+                        return Task.CompletedTask;
+                    }, persistPrepareResultToFile: pathSerialize[0]);
+                    return 0;
+                }, new string[] { path });
+
+                CoverageResult result = TestInstrumentationHelper.GetCoverageResult(path);
+
+                result.Document("Instrumentation.Yield.cs")
+                      .Method("System.Boolean Coverlet.Core.Samples.Tests.Yield/<Three>d__3::MoveNext()")
+                      .AssertLinesCovered((42, 1), (43, 1), (44, 1))
+                      .ExpectedTotalNumberOfBranches(0);
+            }
+            finally
+            {
+                File.Delete(path);
+            }
+        }
+
+        [Fact]
+        public void Yield_Enumerable()
+        {
+            string path = Path.GetTempFileName();
+            try
+            {
+                FunctionExecutor.Run(async (string[] pathSerialize) =>
+                {
+                    CoveragePrepareResult coveragePrepareResult = await TestInstrumentationHelper.Run<Yield>(instance =>
+                    {
+                        foreach (var _ in instance.Enumerable(new[] { "one", "two", "three", "four" })) ;
+
+                        return Task.CompletedTask;
+                    }, persistPrepareResultToFile: pathSerialize[0]);
+                    return 0;
+                }, new string[] { path });
+
+                CoverageResult result = TestInstrumentationHelper.GetCoverageResult(path);
+
+                result.Document("Instrumentation.Yield.cs")
+                      .Method("System.Boolean Coverlet.Core.Samples.Tests.Yield/<Enumerable>d__4::MoveNext()")
+                      .AssertLinesCovered((51, 4))
```

Comment:
can you be more specific with covered lines?i.e. 48,49,50,51,52,53

Reply:
I updated the assert, but line 49 (`foreach (var item in ls)`) got 11 hits which seems too high to me. Is this correct or another bug?

Reply:
Also added extra assertions for release builds.

Reply:
> I updated the assert, but line 49 (foreach (var item in ls)) got 11 hits which seems too high to me. Is this correct or another bug?

Interesting, we should understand why, try to debug test and take a look this part to understand if that row has got 11 hits or is a sum https://github.com/tonerdo/coverlet/blob/master/src/coverlet.core/Coverage.cs#L357

Reply:
As you suspected it's a sum. `foreach (var item in ls)` can have four individual breakpoints:

```csharp
foreach (       // hit 1x on entry
    var item    // hit 4x
    in          // hit 5x
    ls)         // hit 1x on entry
```

Reply:
ok so it's expected!

Reply:
Refactored test a bit to be more clear!